Header header
int32 n
Light[] lights